Index: busybox-1.38.0/libbb/yescrypt/alg-yescrypt-common.c
===================================================================
--- busybox-1.38.0.orig/libbb/yescrypt/alg-yescrypt-common.c
+++ busybox-1.38.0/libbb/yescrypt/alg-yescrypt-common.c
@@ -398,7 +398,7 @@ char *yescrypt_r(
 		goto fail;
  ret:
 	free_region(yctx->local);
-	explicit_bzero(&u, sizeof(u));
+	bzero(&u, sizeof(u));
 	return buf;
  fail:
 	buf = NULL;
Index: busybox-1.38.0/libbb/yescrypt/alg-yescrypt-kdf.c
===================================================================
--- busybox-1.38.0.orig/libbb/yescrypt/alg-yescrypt-kdf.c
+++ busybox-1.38.0/libbb/yescrypt/alg-yescrypt-kdf.c
@@ -219,7 +219,7 @@ static void salsa20(salsa20_blk_t *restr
 	}
 #if 0
 	/* Too expensive */
-	explicit_bzero(&X, sizeof(X));
+	bzero(&X, sizeof(X));
 #endif
 }
 
@@ -1136,7 +1136,7 @@ static int yescrypt_kdf32_body(
 		}
 	}
 
-	explicit_bzero(&u, sizeof(u));
+	bzero(&u, sizeof(u));
 
 	/* Success! */
 	return 0;
@@ -1205,7 +1205,7 @@ int yescrypt_kdf32(
 			passwd, passwdlen,
 			flags, N, t, buf32);
 
-	explicit_bzero(dk32, sizeof(dk32));
+	bzero(dk32, sizeof(dk32));
 
 	dbg("yescrypt_kdf32_body:%d", retval);
 	return retval;
