--- a/test/python/binding/uMediaClient.i	2026-04-20 09:06:14.622602298 +0100
+++ b/test/python/binding/MediaClient.i	2026-04-20 09:06:03.844949527 +0100
@@ -21,8 +21,10 @@
 class MediaPlayer : public uMediaClient {
 public:
 	MediaPlayer() : uMediaClient(false, UMS_CONNECTOR_PRIVATE_BUS), queue_(NULL) {
+#if PY_VERSION_HEX < 0x03070000
 		if (!PyEval_ThreadsInitialized())
 			PyEval_InitThreads();
+#endif
 
 		pthread_create(&in_thr_, NULL, inputThread, this);
 	}
